src/test/java/com/google/sps/servlets/UserDataServletTest.java

```diff
@@ -0,0 +1,228 @@
+  private static final String USER_FOUND_PROPERTY = "user-found";
+  private static final String USER_FRIENDS_LIST_PROPERTY = "friends-list";
+  private static final String USER_ID_PROPERTY = "id";
+  private static final String USER_NAME_PROPERTY = "name";
+
+  // Test User info
+  private static final String TEST_USER_NAME = "Tim";
+  private static final String TEST_USER_ID = "123";
+  private static final String TEST_USER_EMAIL = "tim@gmail.com";
+  private static final String TEST_USER_BIO = "Amazing!";
+  private static final String[] TEST_USER_FRIENDS_LIST = new String[]{"321"};
+  private static final String ALTERNATE_TEST_USER_NAME = "John";
+
+  private final Gson gson = new Gson();
+  // Uses a local datastore stored in memory for tests
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Mock private HttpServletRequest mockRequest;
+  @Mock private HttpServletResponse mockResponse;
+  private DatastoreService datastore;
+  private UserDataServlet servletUnderTest;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    servletUnderTest = new UserDataServlet();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  /**
+   * Tests the doGet method, where the datastore doesn't contain the test user id.
+   *
+   * <p>Expected output: JSON response with "user-found" property as false.
+   */
+  @Test
+  public void testGetMethodWithInvalidId() throws Exception {
+    // Mock the getParameter call
+    when(mockRequest.getParameter(USER_ID_PROPERTY)).thenReturn(TEST_USER_ID);
+
+    // Store output given by mockResponse
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(mockResponse.getWriter()).thenReturn(writer);
+
+    // Test the doGet method by calling the wrapper method
+    servletUnderTest.doGetWrapper(datastore, mockRequest, mockResponse);
+    writer.flush();
+
+    // Check that the user wasn't found in datastore
+    Map<String, Object> actual = gson.fromJson(stringWriter.toString(), Map.class);
+    assertThat(actual).containsEntry(USER_FOUND_PROPERTY, false);
+  }
+
+  /**
+   * Tests the doGet method, where datastore does contain the test user id.
+   *
+   * <p>Expected output: JSON response with "user-found" property as true,
+   * and the user properties filled in with the test user properties.
+   */
+  @Test
+  public void testGetMethodWithValidId() throws Exception {
+    // Mock the getParameter call
+    when(mockRequest.getParameter(USER_ID_PROPERTY)).thenReturn(TEST_USER_ID);
+
+    // Store output given by mockResponse
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(mockResponse.getWriter()).thenReturn(writer);
+
+    addTestUserEntityToDatastore(datastore);
```

Comment:
This probably negates my earlier idea of not having a local data store variable

Reply:
Confused, how can we eliminate the local data store variable? Seems like it is required in the tests.

Reply:
See https://github.com/googleinterns/step48-2020/pull/13#issuecomment-656928605